src/controller/python/BUILD.gn

```diff
@@ -111,21 +113,79 @@ shared_library("ChipDeviceCtrl") {
   configs += [ ":controller_wno_deprecate" ]
 }
 
-pw_python_action("python") {
-  script = "build-chip-wheel.py"
+template("chip_python_wheel_action") {
+  _dist_dir = "${root_out_dir}/controller/python"
+
+  _py_manifest_file = "${target_gen_dir}/${target_name}.py_manifest.json"
+
+  pw_python_action(target_name) {
+    script = "build-chip-wheel.py"
+    forward_variables_from(invoker, "*")
 
-  _py_manifest_files = [
+    _py_manifest_files_rebased = []
+    foreach(_manifest_entry, py_manifest_files) {
+      inputs += _manifest_entry.sources
+      _py_manifest_files_rebased += [
+        {
+          src_dir = rebase_path(_manifest_entry.src_dir,
+                                get_path_info(_py_manifest_file, "dir"))
+          sources = rebase_path(_manifest_entry.sources, _manifest_entry.src_dir)
+        },
+      ]
+    }
+
+    if (defined(invoker.py_scripts)) {
+      _py_scripts = invoker.py_scripts
+    } else {
+      _py_scripts = [ ]
+    }
+
+    _py_manifest = {
+      files = _py_manifest_files_rebased
+      packages = py_packages
+      scripts = _py_scripts
+      package_reqs = py_package_reqs
+    }
+
+    write_file(_py_manifest_file, _py_manifest, "json")
+
+    args = [
+      "--package_name",
+      py_package_name,
+      "--build_number",
+      chip_python_version,
+      "--build_dir",
+      rebase_path("${target_gen_dir}/${target_name}.py_build", root_build_dir),
+      "--dist_dir",
+      rebase_path(_dist_dir, root_build_dir),
+      "--manifest",
+      rebase_path(_py_manifest_file, root_build_dir),
+      "--plat-name",
+      py_platform_tag,
+    ]
+
+    if (defined(invoker.lib_name)) {
+      args += [
+        "--lib-name",
+        lib_name,
+      ]
+    }
+
+    outputs = [ "${_dist_dir}/$output_name" ]
+  }
+}
+
+chip_python_wheel_action("chip-library") {
```

Comment:
Can we re-name this later to `chip-core`, and `chip-clusters`?

Reply:
Works for me. In a follow up or should I add it to this PR?